### ISSUE TITLE
Get-TargetResource for EXODkimSigningConfig prompts credentials

### ISSUE THREAD

**Issue by NikCharlebois** [Microsoft/Office365DSC#72]
#### Details of the scenario you tried and the problem that is occurring
When calling the Get-TargetResource method on the ExoDkimSigningConfig resource, it writes the following message to the host:

Creating a new session for implicit remoting of "Get-DkimSigningConfig" command...

and then prompts for credentials. 

#### Verbose logs showing the problem
"Creating a new session for implicit remoting of "Get-DkimSigningConfig" command..."

#### Suggested solution to the issue
N/A

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
Occurs when calling the Get-TargetResource

#### The operating system the target node is running
N/A

**Comment by NikCharlebois** [comment]
@RobBiddle can you please take a look?

**Comment by RobBiddle** [comment]
@NikCharlebois I haven't been able to reproduce that error.  Get/Test/Set-TargetResource are working using an EXODkimSigningConfig config against my production tenant.

I have seen that error before.  It was happening a lot with all EXO resources prior to adding in additional error handling during the Exchange Online connection process.  Essentially the error means that the cmdlets were imported but the Exchange Online session is no longer in a working state.

Can you email me the full verbose output?

**Comment by NikCharlebois** [comment]
@RobBiddle probably best for you to grab my branch:
https://github.com/NikCharlebois/Office365DSC/tree/GUI-Reverse-Logic

Then run Export-O365Configuration and only select the new resources you've introduced. You'll see the error pop-up. Thanks

**Comment by RobBiddle** [comment]
I pulled your branch.  My configuration still works.
When running Export-O365Configuration selecting only the new EXO Resources it didn't even make it to EXODkimSigningConfig...  When it got to EXOClientAccessRule it prompted me for the Action parameter.  It's a mandatory parameter with no default, I guess I need to assign that a value inside Export-TargetResource for that resource?

**Comment by RobBiddle** [comment]
FYI - After I provided the Action value for EXOClientAccessRule at the console, it continued and made it past EXODkimSigningConfig without prompting for anything.  

It prompted again in EXOHostedContentFilterRule for the mandatory parameter HostedContentFilterPolicy.  And then again in EXOSafeAttachmentRule for the mandatory parameter SafeAttachmentPolicy.  So looks like I need to set some values for those, although I'm not quite sure how that will work with other tenant environments.  Hopefully it's not possible to delete the 'default' policies, if that's the case then 'default' would be a safe value to use.

**Comment by RobBiddle** [comment]
So for SafeLinksPolicy, there is no default policy.  So any default value we would use in Export-TargetResource of EXOSafeLinksRule would likely never exist.  The rest of those could use reasonably expect a policy named 'default' to exist.

**Comment by RobBiddle** [comment]
@NikCharlebois I ran Export-O365Configuration several times selecting all EXO resources except the EXO****Rule resources that are missing parameter values in Export-TargetResource.  It completed each time without any prompts or errors, including EXODkimSigningConfig.

I just tried again and included ALL resources except EXO****Rule.  
It made it past:

O365AdminAuditLogConfig
EXOATPPolicyForO365
EXOCASMailboxPlan
EXODkimSigningConfig
EXOHostedConnectionFilterPolicy
EXOHostedContentFilterPolicy
EXOSafeAttachmentPolicy
EXOSafeLinksPolicy
...but then when it got to **EXOMailboxSettings** I got the "Creating a new session for implicit remoting of "Get-Mailbox" command..." and a credential prompt.  Is it possible that  the problem is occuring due to **Invoke-ExoCommand** ?  I ended up creating **Connect-ExchangeOnline** because I kept running into issues like that using **Invoke-ExoCommand**.
